socs/agents/starcam_lat/agent.py

```diff
@@ -0,0 +1,262 @@
+                  use_HP_bool,
+                  spike_limit_value,
+                  dynamic_hot_pixels_bool,
+                  r_smooth_value,
+                  high_pass_filter_bool,
+                  r_high_pass_filter_value,
+                  centroid_search_border_value,
+                  filter_return_image_bool,
+                  n_sigma_value,
+                  star_spacing_value]
+        # Pack values into the command for the camera
+        self.cmds_for_camera = struct.pack('ddddddfiiiiiiiiiifffffffff',
+                                           *values)
+        # send commands to the camera
+        self.comm.sendto(self.cmds_for_camera, (self.ip, self.port))
+        print("Commands sent to camera")
+        # Return the list of values
+        return values
+
+    def get_astrom_data(self):
+        """
+        get_astrom_data()
+
+        **Process**
+        receives and unpacks data from camera
+
+        **Return**
+        returns dictionary of unpacked data
+        """
+        (scdata_raw, _) = self.comm.recvfrom(224)
+        data = struct.unpack_from("dddddddddddddiiiiiiiiddiiiiiiiiiiiiiifiii",
+                                  scdata_raw)
+        keys = ['c_time',
+                'gmt',
+                'blob_num',
+                'obs_ra',
+                'astrom_ra',
+                'obs_dec',
+                'fr',
+                'ps',
+                'alt',
+                'az',
+                'ir',
+                'astrom_solve_time',
+                'camera_time']
+        # Create a dictionary of the unpacked data
+        astrom_data = [data[i] for i in range(len(keys))]
+        astrom_data_dict = {keys[i]: astrom_data[i] for i in range(len(keys))}
+        return astrom_data_dict
+
+    def close(self):
+        """
+        close()
+
+        **Process**
+        closes the socket of the connection
+        """
+        self.comm.close()
+
+
+class StarcamAgent:
+
+    def __init__(self, agent, ip_address, port):
+        self.agent = agent
+        self.active = True
+        self.log = agent.log
+        self.job = None
+        self.take_data = False
+        self.lock = TimeoutLock()
+        agg_params = {'frame_length': 60}
+        self.agent.register_feed("starcamera", record=True,
+                                 agg_params=agg_params, buffer_time=1)
+        try:
+            self.StarcamHelper = StarcamHelper(ip_address, port)
+        except socket.timeout:
+            self.log.error("Starcamera connection has times out")
+            return False, "Timeout"
+
+    @ocs_agent.param('_')
+    def send_commands(self, session, params=None):
+        """
+        send_commands()
+
+        **Process**
+        packs and sends camera+astrometry-related commands to starcam
+
+        **Return**
+        returns a touple with True/False and a string describing whether
+        or not a lock could be acquired and commands were sent to the sc
+        """
```

Comment:
The syntax of this is still a bit off. Full example can be found [here](https://ocs.readthedocs.io/en/main/developer/agent_references/documentation.html#task-and-process-documentation). It should look like:
```
"""send_commands()

**Process** - Pack and send camera and astrometry related commands to the star camera.

"""
```

We don't include the `Returns:` section in our agent operation docstrings.